src/kv/kv_types.h

```diff
@@ -50,8 +50,58 @@ namespace kv
   DECLARE_JSON_TYPE(TxID);
   DECLARE_JSON_REQUIRED_FIELDS(TxID, term, version)
 
-  using BatchVector = std::vector<
-    std::tuple<kv::Version, std::shared_ptr<std::vector<uint8_t>>, bool>>;
+  // SeqNo indexes transactions processed by the consensus protocol providing
+  // ordering
+  using SeqNo = int64_t;
+
+  struct Configuration
+  {
+    struct NodeInfo
+    {
+      std::string hostname;
+      std::string port;
+      tls::Pem cert = {};
```

Comment:
Depending on a type from `tls` isn't ideal here. I'm aware this was already the case. Could the configuration be defined in a different file, outside of the `kv` namespace?

Reply:
Agreed, it's not clear to me why this needs to be a Pem. Then I had a look and as best I can tell it's not used at all, so I will remove it.

Reply:
Removed